darts/utils/utils.py

```diff
@@ -110,9 +117,84 @@ def fit(self, a, b=0, c=0):
+
+                for param_name, param in signature(method_to_sanitize).parameters.items():
+                    if param.default == Parameter.empty and param.kind != Parameter.VAR_POSITIONAL:
+                        only_kwargs.pop(param_name)
+                    else:
+                        only_args.pop(param_name)
+
+                only_args.pop('self')
+
+                getattr(self, sanity_check_method)(*only_args.values(), **only_kwargs)
+            return method_to_sanitize(self, *only_args.values(), **only_kwargs)
         return sanitized_method
     return decorator
+
+
+def _backtest_general_checks(series, kwargs):
+    """
+    Performs checks common to ForecastingModel and RegressionModel backtest() methods
+
+    Parameters
+    ----------
+    series
+        Either training_series when called from ForecastingModel, or target_series if called from RegressionModel
+    signature_params
+        A dictionary of the signature parameters of the calling method, to get the default values
+        Typically would be signature(self.backtest).parameters
+    kwargs
+        Params specified by the caller of backtest(), they take precedence over the arguments' default values
+    """
+
+    # parse kwargs
+    n = SimpleNamespace(**kwargs)
+
+    # check forecast horizon
+    forecast_horizon = n.forecast_horizon
+
+    raise_if_not(forecast_horizon > 0, 'The provided forecasting horizon must be a positive integer.', logger)
+
+    # check start parameter
+    if hasattr(n, 'start'):
+        if isinstance(n.start, float):
+            raise_if_not(n.start >= 0.0 and n.start < 1.0, '`start` should be between 0.0 and 1.0.', logger)
+        elif isinstance(n.start, pd.Timestamp):
+            raise_if(n.start not in series, '`start` timestamp must be an entry in the time series\' time index')
+            raise_if(n.start == series.end_time(), '`start` timestamp is the last timestamp of the series', logger)
+        elif isinstance(n.start, int):
+            raise_if_not(n.start >= 0, logger)
+            raise_if(n.start > len(series), '`start` index should be smaller than length of the series', logger)
+        else:
+            raise_log(TypeError("`start` needs to be either `float`, `int` or `pd.Timestamp`"), logger)
+
+    start = _backtest_convert_start(n.start, series)
+
+    raise_if(start == series.start_time(), '`start` corresponds to the first timestamp of the series, '
+             'resulting in empty training set')
+
+    # check that trim_to_series and start together form a valid combination
+    trim_to_series = n.trim_to_series
+
+    if trim_to_series:
+        raise_if_not(start + series.freq() * forecast_horizon in series,
+                     '`start` timestamp is too late in the series to make any predictions with'
+                     '`trim_to_series` set to `True`.', logger)
+
+
+def _backtest_convert_start(start: Union[pd.Timestamp, float, int], training_series: TimeSeries) -> pd.Timestamp:
```

Comment:
I think this method might also be useful outside of backtesting as well. You could rename it to reflect this. For instance to something like `get_ts_at_point(point: Union[pd.Timestamp, float, int], series: TimeSeries) -> pd.Timestamp`.

In fact, I think later on (not for this PR) we could even think of having such "fancy indexing" natively supported in `TimeSeries`, as well as in some `TimeSeries` methods such as `split_after`, `split_after`, `slice`, etc.

Reply:
I think so as well.

Being able to at least split a `TimeSeries` according to some ratio or index would indeed be really nice, and this method could help there.